### ISSUE TITLE
CipherSuites.select/1 fails on ECDHE-ECDSA-CHACHA20-POLY1305 and ECDHE-RSA-CHACHA20-POLY1305

### ISSUE THREAD

**Issue by DavidAntaramian** [voltone/cipher_suites#1]
First, thanks for this library. It helps a lot with integration!

I'm configuring an application to follow Mozilla's "modern" [Server Side TLS guidelines](https://wiki.mozilla.org/Security/Server_Side_TLS) which recommends the following ciphersuites in OpenSSL notation:

```
ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256
```

I've tried to convert that to a set of Erlang ciphersuites using CipherSuites:

```elixir
CipherSuites.select("ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256")
```

This results in an error, though:

```
** (FunctionClauseError) no function clause matching in CipherSuites.find/1                                                                                                                                                                                                                                                                                                                 
    The following arguments were given to CipherSuites.find/1:
    
        # 1
        "ECDHE-ECDSA-CHACHA20-POLY1305"
    
    Attempted function clauses (showing 10 out of 70):
    
        defp find("ALL")
        defp find("COMPLEMENTOFDEFAULT")
        defp find("COMPLEMENTOFALL")
        defp find("HIGH")
        defp find("MEDIUM")
        defp find("LOW")
        defp find("eNULL")
        defp find("NULL")
        defp find("aNULL")
        defp find("kRSA")
    
    (cipher_suites) lib/cipher_suites.ex:206: CipherSuites.find/1
    (cipher_suites) lib/cipher_suites.ex:202: CipherSuites.find/2
    (cipher_suites) lib/cipher_suites.ex:168: CipherSuites.append/2
    (elixir) lib/enum.ex:1899: Enum."-reduce/3-lists^foldl/2-0-"/3
    (cipher_suites) lib/cipher_suites.ex:120: CipherSuites.filter/1
```

This section seems to be the culprit:

```
ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305
```

Is this a bug, or did I have incorrect assumptions about the library's functionality?

Versions:
- Elixir 1.6.3
- Erlang 20.2.2
- CipherSuite 0.2.1
- macOS 10.13.3

**Comment by voltone** [comment]
The issue here is that OTP's `:ssl_cipher` module does not define a mapping from this particular OpenSSL-style name ("ECDHE-ECDSA-CHACHA20-POLY1305") to the corresponding binary ID:
https://github.com/erlang/otp/blob/master/lib/ssl/src/ssl_cipher.erl#L1880

I didn't want to replicate that list in the library, but perhaps I can add a list of additional mappings to check if `:ssl_cipher` comes back empty-handed. Or at least improve the error handling (print a warning and ignore the entry?).

**Comment by voltone** [comment]
Fixed in v0.3.0

**Comment by DavidAntaramian** [comment]
Thank you!
